src/Chapter3.hs

```diff
@@ -971,6 +1057,28 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Show, Eq, Ord, Enum)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend d = d >= Saturday
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay Sunday = Monday
+nextDay d = succ d
+
+daysToParty :: DayOfWeek -> Int
+daysToParty d
+  | d <= Friday = (fromEnum Friday) - (fromEnum d)
+  | otherwise = 7 + (fromEnum Friday) - (fromEnum d)
```

Comment:
You could also just use `mod`. Don't worry, mod works as needed on negatives as well:

```haskell
> mod (-1) 7
6
```